### PR TITLE
Upgrade to Glean v51

### PR DESCRIPTION
**Pull request by badboy**
Upgrades Glean to v51 and includes #11344 to bring them in sync.

[v50 was a major release](https://github.com/mozilla/glean/releases/tag/v50.0.0).
v51 dropped some deprecated functionality. We never deployed v50 on iOS, so we just skip the deprecated stuff and remove usage of them right away here.

cc @travis79

### CODE REVIEW

**Comment by badboy** [comment]
@tarikeshaq [the a-s release says](https://github.com/mozilla/rust-components-swift/releases/tag/93.7.0):

> Added `MOZ_APPSERVICES_MODULE` environment variable to specify the megazord module for iOS (#5042). If it is missing, no module is imported.

Do we need to do anything here?

File: Storage/Rust/RustLogins.swift [hunk 1]
```diff
@@ -501,8 +501,8 @@ public class RustLogins {
 
         queue.async {
             guard self.isOpen else {
-                let error = LoginsStoreError.MismatchedLock(message: "Database is closed")
-                deferred.fill(Maybe(failure: error as MaybeErrorType))
+                //let error = LoginsStoreError.MismatchedLock(message: "Database is closed")
```
Comment (badboy) [inline Storage/Rust/RustLogins.swift:504]: @tarikeshaq the changelog says:

> Note that some errors were removed, which is technically a breaking change, but none of our consumers use those errors so it's not a breaking change in practice.

Guess that wasn't really correct. It's in use here and fails the build now. I have no idea what to do, can you provide the changes needed here?

**Comment by tarikeshaq** [comment]
Thanks a ton @badboy and apologies that the changelog was incorrect about the logins errors 🙏 ! cc @mhammond @bendk 

I can take it from here and submit a PR with your changes and a commit on top.

**Comment by tarikeshaq** [comment]
I opened #11393, it should be safe to close this in favour of that PR!

**Comment by lmarceau** [comment]
Thanks @tarikeshaq, closing this PR then 👍

**Pull request closed without merging**